### DISCUSSION TITLE
Disabled fields with yup schema running validation checks

### DISCUSSION THREAD

**Discussion by reintroducing** [react-hook-form/react-hook-form#6823] [Q&A]
Full disclosure, this is my first time messing with react-hook-form, but I've combed the docs and Googled furiously and could not find an answer.

I've set up a CodeSandbox that shows the issue: https://codesandbox.io/s/react-hook-form-v7-validationschema-forked-i4tqs?file=/src/index.js

Pressing submit runs the validation check on the `firstName` field. I would expect that disabled fields would not run validation (and would not be part of the submitted data either, like it isn't in regular HTML forms, although that seems to be a hotly debated topic for this library). Can someone please enlighten me on how to set disabled fields to not only not validate (even if their validation says `required` in the schema) but also not submit in the final form data?

As a reference, here is the form in plain ole HTML, acting as expected: https://codesandbox.io/s/vigilant-wright-pw12r?file=/index.html

**Comment by bluebill1049** [comment]
`disabled` is limited to build-in validation, for resolver you can consider using context objects to optional update your schema, or conditional set your schema validation based on `undefined` value.

**Reply by bluebill1049** [reply]
i will enhance the doc to highlight this.

**Reply by reintroducing** [reply]
@bluebill1049 If i'm understanding you correctly, the first option to use context objects would look something like this? https://codesandbox.io/s/react-hook-form-v7-validationschema-forked-o3khk?file=/src/index.js

The problem with that approach is that you would have to manage the boolean for every disabled field in the parent, not ideal if there are a lot of disabled fields. Is there a cleaner way to do this that I'm not seeing? 

As for the second option, `conditional set your schema validation based on undefined value`, what would that look like? Is there an example somewhere I can take a look at? Even though I don't know what the code would look like, when thinking through your statement in my mind, wouldn't i still need to know if the field is disabled or not, because a field could also be undefined and not disabled, in which case i may still want to run the validation on it, no?

**Reply by iliakut** [reply]
> @bluebill1049 If i'm understanding you correctly, the first option to use context objects would look something like this? https://codesandbox.io/s/react-hook-form-v7-validationschema-forked-o3khk?file=/src/index.js
> 
> The problem with that approach is that you would have to manage the boolean for every disabled field in the parent, not ideal if there are a lot of disabled fields. Is there a cleaner way to do this that I'm not seeing?
> 
> As for the second option, `conditional set your schema validation based on undefined value`, what would that look like? Is there an example somewhere I can take a look at? Even though I don't know what the code would look like, when thinking through your statement in my mind, wouldn't i still need to know if the field is disabled or not, because a field could also be undefined and not disabled, in which case i may still want to run the validation on it, no?


Thank you for this example, it helped me to understand how to make it works. I've siplified your codesandbox a bit
here is my working solutions without context. You only need register checkbox and use when in your validation: 
https://codesandbox.io/s/react-hook-form-v7-validationschema-forked-4mb1f?file=/src/index.js
